java-checks/src/main/java/org/sonar/java/checks/UselessConditionCheck.java

```diff
@@ -73,7 +72,16 @@ public void visitMethod(MethodTree tree) {
   }
 
   private void raiseIssue(Tree tree, String value) {
-    context.addIssue(tree, this, String.format("Change this condition so that it does not always evaluate to \"%s\"", value));
+    addIssue(tree, String.format("Change this condition so that it does not always evaluate to \"%s\"", value));
+  }
+
+  private class Check extends SymbolicExecutionCheck {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Should org.sonar.java.checks.UselessConditionCheck$Check be a _static_ inner class? [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASIC_INNER_SHOULD_BE_STATIC)